### ISSUE TITLE
ccv0: ccv0.sh script can fail when run as non-root user

### ISSUE THREAD

**Issue by stevenhorsman** [kata-containers/kata-containers#4241]
This issue is related to the problems seen in https://github.com/kata-containers/tests/issues/4692 where git throws an error when it encounters repo owned by another user.

Due to `rootfs.sh` needing to being run as `sudo` and earlier parts of the script checking out the tests repo as the standard user it's possible to get the following error when running: `~/ccv0.sh create_rootfs`:
```bash
/home/stevenh/go/src/github.com/kata-containers/tests /
fatal: unsafe repository ('/home/stevenh/go/src/github.com/kata-containers/tests' is owned by someone else)
To add an exception for this directory, call:

	git config --global --add safe.directory /home/stevenh/go/src/github.com/kata-containers/tests
Failed at 591: ${script_dir}/../../../ci/install_libseccomp.sh "${libseccomp_install_dir}" "${gperf_install_dir}"
[ccv0.sh:259] ERROR: sudo -E OS_VERSION= GOPATH=/home/stevenh/go EXTRA_PKGS=vim iputils-ping net-tools DEBUG= USE_DOCKER=1 SKOPEO= AA_KBC= UMOCI=yes SECCOMP=yes ./rootfs.sh -r /home/stevenh/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/rootfs ubuntu
```

I think it may have been silently failing for a while, but with the change to add in `set -o errexit` it causes the script to stop.

**Comment by stevenhorsman** [comment]
Closed with #4242 merging

**Comment by stevenhorsman** [comment]
Re-opening this issue. Now the problem of the tests repo ownership has been solved the kata-containers one is complaining about the same thing. Unlike the tests repo check out it doesn't seem to block the script, but it's still worth fixing

**Comment by stevenhorsman** [comment]
The problem with doing the same fix as I did with the tests repo and make it owned by root is that a bunch of build process write to the kata-containers directory (eg building the runtime) and I don't really want to make every process run as root. I'll try some alternatives first before I fall back to that.

**Comment by stevenhorsman** [comment]
After a few attempts I managed to avoid changing the whole kata-containers repo directory to be owned by root by re-using some code `add_repo_to_git_safe_directory` that does the functionality I wanted in [kata-containers/kata-containers#4329](https://github.com/kata-containers/kata-containers/pull/4329). The dependency chain and ownership was still a bit funny, but I managed to update it after the check-out and it's working for me manually.

**Comment by stevenhorsman** [comment]
Closed with #4329 merging
